### ISSUE TITLE
Explicit-AD Arnoldi precheck under-reports rho(J^T) on complex tensors: a stale duplicate of arnoldi_spectral_radius uses real buffers and an unconjugated dot

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#828]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

Verified live from the #802 backlog table (row: #334 `algorithms/ad_utils.py:664`, "Preserve complex dtype in Arnoldi basis buffers"). Reproduced on `main` at `32907d0`.

## There are two copies of `arnoldi_spectral_radius`

The fix landed in a **new module** and the buggy original was never deleted:

| | buffers | orthogonalisation | used by |
|---|---|---|---|
| `algorithms/_arnoldi.py:13` | `jnp.zeros(..., dtype=v0.dtype)`, complex-preserving | `jnp.vdot` (Hermitian) | implicit AD — `_ctm_energy_ad.py:1477`, `_ctm_honeycomb_ad.py:408` |
| `algorithms/ad_utils.py:192` | `jnp.zeros((n, k + 1))` — **real** | `jnp.dot` — **unconjugated** | explicit AD — `_ctm_tensor_converge_bwd`, `ad_utils.py:896` |

So the implicit-AD path is correct and the explicit-AD path is not.

## Reproduction

```
true rho                     = 2.0
_arnoldi.py (implicit AD)    = 2.000000000000
ad_utils.py (explicit AD)    = 0.972398244785   <- under by 2.06x

  ComplexWarning: Casting complex values to real discards the imaginary part
  FutureWarning: scatter inputs have incompatible types: cannot safely cast
                 value from dtype=complex128 to dtype=float64 ... In future
                 JAX releases this will result in an error.

real control: fixed=2.000000  stale=2.000000  (both correct)
```

Operator is `diag(2i, 1, 0.5)`, so the spectral radius is unambiguous. Script at the bottom.

**Real inputs are unaffected** — `jnp.zeros` is fine and `dot == vdot` when everything is real. This is complex-only, which is why it has survived: every real-tensor run exercises the function correctly.

## Why it matters

The caller is the **divergence precheck** for the explicit-AD CTM backward:

```python
rho = arnoldi_spectral_radius(_flat_matvec, g_flat, n_iter=20)
arnoldi_threshold = getattr(config, "adjoint_arnoldi_threshold", 5.0)
if rho >= arnoldi_threshold:
    raise CTMRGGradientError(spectral_radius=rho)
```

`adjoint_arnoldi_precheck` defaults to `True`, so this runs by default. Its whole job is to refuse a gradient when `rho(Jᵀ)` is too large for the adjoint fixed point to converge — and on complex tensors it under-reports by ~2x in this example, taking a `rho = 2.0` operator to a reported `0.972`, i.e. from clearly divergent to comfortably below even the `rho >= 1` line. The guard that exists to catch a divergent adjoint is the thing that fails.

Complex `DenseTensor` support was fixed rather than gated (#722, via #724), so this is reachable rather than theoretical.

## Two defects, and the second one matters for the record

1. **Real buffers** — `Q = jnp.zeros((n, k + 1))` and `H = jnp.zeros((k + 1, k))`. `Q.at[:, 0].set(q)` then casts the complex Krylov vector down, discarding the imaginary part. This is what the original comment named.
2. **Unconjugated inner product** — `h = jnp.dot(Q[:, i], v)`. Modified Gram-Schmidt needs `⟨q_i, v⟩ = conj(q_i)·v`; `jnp.dot` gives the bilinear form, so the projection is wrong even setting the dtype aside.

Defect 2 is the **same class as #235/#330**, which #802 records as *closed*: "`_tree_dot` is now Hermitian (`Re<conj(a),b>`) and tangent projection uses `jnp.vdot` — the complex-inner-product cluster (#235, #330) is closed." That cluster was cleared in `ipeps_optimize.py` and `pess_optimize.py`; this site was missed because it is a duplicate implementation nobody grepped for.

## Suggested fix

Delete `ad_utils.arnoldi_spectral_radius` and import from `_arnoldi` at the single call site. The two functions have the same signature, and `_arnoldi.py` is already the reviewed one — it preserves dtype, uses `jnp.vdot`, guards breakdown with `jnp.where` rather than a Python `break` on a traced value, and forces `eigvals` to CPU (cuSOLVER `geev` fails on small complex matrices). Keeping two copies is what produced this.

Worth a grep for other duplicated numerical helpers at the same time — the failure mode here is not the bug, it is that a fix landed on one of two copies and the cluster was then marked closed.

<details><summary>Reproduction script</summary>

```python
import jax, jax.numpy as jnp
jax.config.update("jax_enable_x64", True)
from tenax.algorithms._arnoldi import arnoldi_spectral_radius as fixed
from tenax.algorithms.ad_utils import arnoldi_spectral_radius as stale

A = jnp.diag(jnp.array([2j, 1.0 + 0j, 0.5 + 0j]))   # rho = 2
v0 = jnp.array([1.0 + 1j, 0.3 - 0.2j, 0.1 + 0.4j])
mv = lambda v: A @ v

print("true rho :", 2.0)
print("_arnoldi :", fixed(mv, v0, n_iter=3))   # 2.0
print("ad_utils :", stale(mv, v0, n_iter=3))   # 0.972...
```

</details>

Refs #802, #334, #235, #330, #722, #724.
